assets/js/favorites.js

```diff
@@ -0,0 +1,40 @@
+var favoritesList;
+
+function init() {
+    favoritesList = localStorage.getItem("favorites");
+    if (favoritesList) {
+        favoritesList = JSON.parse(favoritesList);
+    } else {
+        favoritesList = [];
+    }
+}
+
+function displayFavorites() {
+    var favoritesContainer = document.getElementById("favorites-list");
+    favoritesContainer.innerHTML = "";
+
+    if (favoritesList.length > 0) {
+        favoritesList.forEach(function (songIndex) {
+            var song = lastSearch[songIndex].track;
+            var listItem = document.createElement("li");
+            listItem.textContent = song.track_name + " by " + song.artist_name;
+            favoritesContainer.appendChild(listItem);
+        });
+    } else {
+        favoritesContainer.innerHTML = "You have no favorite songs yet.";
+    }
+}
+
+function addToFavorites(songIndex) {
```

Comment:
naming the parameter as songIndex seems to imply that is a number, but it looks like you're planning on storing song objects in here